src/context/filter.rs

```diff
@@ -0,0 +1,155 @@
+    pub fn args_count(&self) -> c_int {
+        unsafe { RedisModule_CommandFilterArgsCount.unwrap()(self.inner) }
+    }
+
+    /// wrapper for RedisModule_CommandFilterArgGet
+    pub fn arg_get(&self, pos: c_int) -> *mut RedisModuleString {
+        unsafe { RedisModule_CommandFilterArgGet.unwrap()(self.inner, pos) }
+    }
+
+    /// wrapper to get argument as a Result<&str, Utf8Error> instead of RedisModuleString
+    pub fn arg_get_as_str<'a>(&self, pos: c_int) -> Result<&'a str, Utf8Error> {
+        let arg = self.arg_get(pos);
+        ValkeyString::from_ptr(arg)
+    }
+
+    /// wrapper to get 0 argument, the command which is always present and return as &str
+    pub fn cmd_get_as_str(&self) -> &str {
+        let cmd = self.arg_get(0);
+        ValkeyString::from_ptr(cmd).unwrap()
+    }
+
+    /// wrapper to get Vector of all args minus the command (0th arg)
+    pub fn get_all_args_wo_cmd(&self) -> Vec<&str> {
+        let mut output = Vec::new();
+        for pos in 1..self.args_count() {
+            match self.arg_get_as_str(pos) {
+                Ok(arg) => output.push(arg),
+                Err(_) => continue, // skip invalid args
+            }
+        }
+        output
+    }
+
+    /// wrapper for RedisModule_CommandFilterArgReplace, accepts simple &str and casts it to *mut RedisModuleString
+    pub fn arg_replace(&self, pos: c_int, arg: &str) {
+        unsafe {
+            RedisModule_CommandFilterArgReplace.unwrap()(
+                self.inner,
+                pos,
+                self.create_module_string(arg),
+            )
+        };
+    }
+
+    /// wrapper for RedisModule_CommandFilterArgInsert, accepts simple &str and casts it to *mut RedisModuleString
+    pub fn arg_insert(&self, pos: c_int, arg: &str) {
+        unsafe {
+            RedisModule_CommandFilterArgInsert.unwrap()(
+                self.inner,
+                pos,
+                self.create_module_string(arg),
+            )
+        };
+    }
+
+    /// wrapper for RedisModule_CommandFilterArgDelete
+    pub fn arg_delete(&self, pos: c_int) {
+        unsafe { RedisModule_CommandFilterArgDelete.unwrap()(self.inner, pos) };
+    }
+
+    /// wrapper for RedisModule_CommandFilterGetClientId, not supported in Redis 7.0
+    #[cfg(all(any(
+        feature = "min-redis-compatibility-version-7-2",
+        feature = "min-valkey-compatibility-version-8-0"
+    ),))]
+    pub fn get_client_id(&self) -> u64 {
+        unsafe { RedisModule_CommandFilterGetClientId.unwrap()(self.inner) }
+    }
+
+    // create a RedisModuleString from a &str
+    fn create_module_string(&self, arg: &str) -> *mut RedisModuleString {
+        let arg_cstring = CString::new(arg).unwrap();
+        let arg_module_string = unsafe {
+            RedisModule_CreateString.unwrap()(
+                null_mut(),
+                arg_cstring.as_ptr(),
+                arg_cstring.as_bytes().len(),
+            )
+        };
+        arg_module_string
+    }
```

Comment:
This function is not needed as we can use existing APIs build around ValkeyString for this. I added a suggestion above

Reply:
updated the code